src/Mvc/MvcManager.php

```diff
@@ -34,113 +39,127 @@
-
-            call_user_func_array([$controller, $action], self::getArgs([$controller, $action]));
-
-            if (method_exists($controller, '__after')) {
-                call_user_func_array([$controller, '__after'], self::getArgs([$controller, '__after']));
-            }
-        }
-    }
-
-    /**
-     * Get Controller
-     * @return QtController
-     * @throws DiException
-     * @throws ReflectionException
-     * @throws ControllerException
-     */
-    private static function getController(): QtController
-    {
-        $fs = Di::get(FileSystem::class);
-
-        $controllerPath = modules_dir() . DS . current_module() . DS . 'Controllers' . DS . current_controller() . '.php';
-
-        if (!$fs->exists($controllerPath)) {
-            throw ControllerException::controllerNotFound(current_controller());
-        }
-
-        require_once $controllerPath;
-
-        $controllerClass = '\\Modules\\' . current_module() . '\\Controllers\\' . current_controller();
-
-        if (!class_exists($controllerClass, false)) {
-            throw ControllerException::controllerNotDefined(current_controller());
-        }
-
-        return new $controllerClass();
-    }
-
-    /**
-     * Get Action
-     * @param QtController $controller
-     * @return string|null
-     * @throws ControllerException
-     */
-    private static function getAction(QtController $controller): ?string
-    {
-        $action = current_action();
-
-        if ($action && !method_exists($controller, $action)) {
-            throw ControllerException::actionNotDefined($action);
-        }
-
-        return $action;
-    }
-
-    /**
-     * Get arguments
-     * @param callable $callable
-     * @return array
-     * @throws DiException
-     * @throws ReflectionException
-     */
-    private static function getArgs(callable $callable): array
-    {
-        return Di::autowire($callable, self::routeParams());
-    }
-
-    /**
-     * Gets the route parameters
-     * @return array
-     */
-    private static function routeParams(): array
-    {
-        return array_map(function ($param) {
-            return $param['value'];
-        }, route_params());
-    }
+	/**
+	 * @param Request $request
+	 * @param Response $response
+	 * @return void
+	 * @throws ConfigException
+	 * @throws ControllerException
+	 * @throws CryptorException
+	 * @throws CsrfException
+	 * @throws DiException
+	 * @throws MiddlewareException
+	 * @throws ReflectionException
+	 * @throws DatabaseException
+	 * @throws LangException
+	 * @throws SessionException
+	 */
+	public static function handle(Request $request, Response $response)
+	{
+		if (current_middlewares()) {
+			list($request, $response) = (new MiddlewareManager())->applyMiddlewares($request, $response);
+		}
+
+		$callback = route_callback();
+		$cacheSettings = route_cache_settings();
+
+		if ($cacheSettings &&
+			session()->getId() &&
+			ViewCache::exists(route_uri(), session()->getId(), $cacheSettings['ttl'])
```

Comment:
`ViewCache::exists` creates instance inside the method, and the next instance is created few lines bellow.
It leads to apply singleton approach